core/src/main/scala/org/apache/spark/api/python/PythonRDD.scala

```diff
@@ -202,7 +202,7 @@ private[spark] class PythonRDD(
       this.interrupt()
     }
 
-    override def run(): Unit = Utils.logUncaughtExceptions {
+    override def run(): Unit = Utils.tryLog {
```

Comment:
The `run` can not have return value. The InterruptedException could happen in any place (when the task is canceled), so I think it's better to do it here.